utils/utils.go

```diff
@@ -69,15 +68,7 @@ func GetKubeClient() *kubernetes.Clientset {
 // ToISO8601DateTimeString converts dateTime to ISO8601 Format
 // ISO8601 Format: 2020-01-01T01:01:01.10101Z.
 func ToISO8601DateTimeString(dateTime time.Time) string {
-	t := dateTime.UTC()
-
-	year, month, day := t.Date()
-	hour, minute, second := t.Clock()
-	micros := t.Nanosecond() / 1000
-
-	return fmt.Sprintf(
-		"%04d-%02d-%02dT%02d:%02d:%02d.%03dZ",
-		year, month, day, hour, minute, second, micros)
+	return dateTime.UTC().Format("2006-01-02T15:04:05.999999Z")
```

Comment:
This method was actually wrong. I've updated the test scenarios to reproduce a bug exposed via a flaky test (time.Now() makes these type of tests flaky).